lib/class-wp-rest-themes-controller.php

```diff
@@ -0,0 +1,251 @@
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function get_items( $request ) {
+		// Retrieve the list of registered collection query parameters.
+		$registered = $this->get_collection_params();
+		$themes     = array();
+
+		if ( isset( $registered['status'], $request['status'] ) && in_array( 'active', $request['status'], true ) ) {
+			$active_theme = wp_get_theme();
+			$active_theme = $this->prepare_item_for_response( $active_theme, $request );
+			$themes[]     = $this->prepare_response_for_collection( $active_theme );
+		}
+
+		$response = rest_ensure_response( $themes );
+
+		$response->header( 'X-WP-Total', count( $themes ) );
+		$response->header( 'X-WP-TotalThemes', count( $themes ) );
+
+		return $response;
+	}
+
+	/**
+	 * Prepares a single theme output for response.
+	 *
+	 * @since 5.0.0
+	 *
+	 * @param WP_Theme        $theme   Theme object.
+	 * @param WP_REST_Request $request Request object.
+	 * @return WP_REST_Response Response object.
+	 */
+	public function prepare_item_for_response( $theme, $request ) {
+		$data   = array();
+		$fields = $this->get_fields_for_response( $request );
+
+		if ( in_array( 'theme_supports', $fields, true ) ) {
+			$formats                           = get_theme_support( 'post-formats' );
+			$formats                           = is_array( $formats ) ? array_values( $formats[0] ) : array();
+			$formats                           = array_merge( array( 'standard' ), $formats );
+			$data['theme_supports']['formats'] = $formats;
+
+			$data['theme_supports']['post-thumbnails'] = false;
+			$post_thumbnails                           = get_theme_support( 'post-thumbnails' );
+
+			if ( $post_thumbnails ) {
+				// $post_thumbnails can contain a nested array of post types.
+				// e.g. array( array( 'post', 'page' ) ).
+				$data['theme_supports']['post-thumbnails'] = is_array( $post_thumbnails ) ? $post_thumbnails[0] : true;
+			}
+		}
+
+		$data = $this->add_additional_fields_to_object( $data, $request );
+
+		// Wrap the data in a response object.
+		$response = rest_ensure_response( $data );
+
+		$response->add_links( $this->prepare_links( $theme ) );
+
+		/**
+		 * Filters theme data returned from the REST API.
+		 *
+		 * @since 5.0.0
+		 *
+		 * @param WP_REST_Response $response The response object.
+		 * @param WP_Theme         $theme    Theme object used to create response.
+		 * @param WP_REST_Request  $request  Request object.
+		 */
+		return apply_filters( 'rest_prepare_theme', $response, $theme, $request );
+	}
+
+	/**
+	 * Prepares links for the theme request.
+	 *
+	 * @since 5.0.0
+	 *
+	 * @param WP_Theme $theme Theme object.
+	 * @return array Links for the given theme.
+	 */
+	protected function prepare_links( $theme ) {
+		$links = array(
+			'self' => array(
+				'href' => rest_url( sprintf( '%s/%s?status=active', $this->namespace, $this->rest_base ) ),
```

Comment:
I think `href` is supposed to be `/wp/v2/themes/<slug>`, which we don't have right now. We could probably remove `prepare_links()` entirely.